src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs

```diff
@@ -922,6 +1034,32 @@ static ImmutableArray<IConfigurationFixProvider> GetConfigurationFixProviders(Im
             }
         }
 
+        private static ImmutableDictionary<LanguageKind, Lazy<ImmutableArray<CopilotCodeFixProvider>>> GetCopilotCodeFixProvidersPerLanguageMap(
+            ImmutableDictionary<string, ImmutableArray<Lazy<CodeFixProvider, CodeChangeProviderMetadata>>> fixersPerLanguageMap)
+        {
+            var fixerMap = ImmutableDictionary.CreateBuilder<LanguageKind, Lazy<ImmutableArray<CopilotCodeFixProvider>>>();
+            foreach (var (language, lazyFixers) in fixersPerLanguageMap)
+            {
+                var lazyFixProviders = new Lazy<ImmutableArray<CopilotCodeFixProvider>>(() => GetFixProviders(lazyFixers));
+                fixerMap.Add(language, lazyFixProviders);
+            }
+
+            return fixerMap.ToImmutable();
+
+            static ImmutableArray<CopilotCodeFixProvider> GetFixProviders(ImmutableArray<Lazy<CodeFixProvider, CodeChangeProviderMetadata>> languageKindAndFixers)
+            {
+                using var builderDisposer = ArrayBuilder<CopilotCodeFixProvider>.GetInstance(out var builder);
+                var orderedLanguageKindAndFixers = ExtensionOrderer.Order(languageKindAndFixers);
+                foreach (var languageKindAndFixersValue in orderedLanguageKindAndFixers)
+                {
+                    if (languageKindAndFixersValue.Value is CopilotCodeFixProvider copilotFixer)
+                        builder.Add(copilotFixer);
+                }
+
+                return builder.ToImmutable();
+            }
+        }
```

Comment:
lots more code that is just not explaiend.

Reply:
The code added here matches the method just above this one, which deals with special fix provider type for suppressions/configurations.